### DISCUSSION TITLE
Thinking about read-only replication

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#5182] [Q&A]
I'm thinking through what it would take to put read-only PocketBase instances on the edge for regional availability.

* PocketBase versions would need to match exactly
* LiteFS data.db
* do not sync logs.db
* Migrations disabled
* Use s3 for file storage
* Sync JS hooks and pb_static
* Disable cron
* Disable backups
* GET served from edge
* PUT/POST/DELETE proxied to origin
* All /api/realtime/* proxied to origin (PocketBase doesn't monitor SQLite for events)
* No db writes allowed in GET JS hooks (could just be convention)
* Not sure what could be done about race conditions where an edge GET following an origin POST when the write hasn't sync'd to the edges yet

Anything else?

Also thinking about a `--readonly` switch that blocks PocketBase from making any writes to the db, for added safety.

**Comment by moussa32** [comment]
Great idea to think about it,
But how do you will manage the updated data in the origin instance being updated in edge nodes ?

**Reply by benallfree** [reply]
LiteFS as mentioned :)

**Comment by pnmcosta** [comment]
Good luck. I haven't had much luck with maintaining consistency a long time using file based db replication, when it goes wrong, it can really go wrong! 😅

I'm keener on a eventually consistent edge of PB instances, I think that a good event sourcing implementation wrapper around collections and records events would work for me. My use case is regional availability where I don't really mind one being behind for a moment as long as they eventually are.

**Reply by benallfree** [reply]
Yes, this is predicated on LiteFS actually working. But fly seems to be using it consistently, so maybe?

**Reply by ganigeorgiev** [reply]
_Side-note: @benallfree While I don't recommend using PocketBase for such use cases, note that if you are planning to use LiteFS only for syncing the single `pb_data/data.db` file then it may not work due to its being FUSE based since it may require mounting the entire directory. As an alternative, another replication tool could be [Marmot](https://github.com/maxpert/marmot) (make sure to check also their [Limitations section](https://github.com/maxpert/marmot?tab=readme-ov-file#limitations))._
